brillig_vm/src/opcodes.rs

```diff
@@ -4,10 +4,25 @@ use serde::{Deserialize, Serialize};
 
 pub type Label = usize;
 
+/// Lays out various ways we might interpret memory when passing to a foreign call external from Brillig.
+///
+/// While we are usually agnostic to how memory is passed within Brillig,
+/// this needs to be encoded somehow when dealing with an external system.
+/// For simplicity, the extra type information is given right in the ForeignCall instructions.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Copy)]
-pub enum RegisterValueOrArray {
+pub enum RegisterOrMemory {
+    /// An immediate value passed to or from an external call
+    /// For a foreign call input, this is read directly from the register.
+    /// For a foreign call output, this is written directly to the register.
```

Comment:
```suggestion
    /// For a foreign call output, the value is written directly to the register.
```

Reply:
This comment and the ones below on the enum variants are just nits that I thought made things a bit clearer